### PR TITLE
Implement Role#getTags and add CacheFlag#ROLE_TAGS

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This introduces the new tags for roles which can be used to identify which role is used for boosters or the bot.

### CODE REVIEW

**Comment by MinnDevelopment** [comment]
This feature hasn't been fully deployed yet and is not yet documented. I will keep this on freezer in the meantime.

**Comment by Andre601** [comment]
If needed could I make a PR targeting this branch, to implement comments.

**Comment by kantenkugel** [comment]
Feel free to add me as reviewer once its ready to be merged. BTW: the last 2 changed files don't seem to be related to the tags at all. Do you want to implement them in some different branch, if not already planned?

File: src/main/java/net/dv8tion/jda/api/entities/Guild.java [hunk 1]
```diff
@@ -1570,6 +1570,46 @@ default Role getRoleById(long id)
...
+            stream.filter(role -> role.getTags().getBotIdLong() == userId)
+                  .findFirst()
+                  .orElse(null)
+        );
+    }
+
+    @Nullable
+    default Role getRoleByBot(@Nonnull String userId)
+    {
+        return getRoleById(MiscUtil.parseSnowflake(userId));
```
Comment (Andre601) [inline src/main/java/net/dv8tion/jda/api/entities/Guild.java:1587]: Is that a typo, or why does this method target getRoleById?

File: src/main/java/net/dv8tion/jda/api/entities/Guild.java [hunk 2]
```diff
@@ -1570,6 +1571,116 @@ default Role getRoleById(long id)
...
+     *
+     * @param  userId
+     *         The user id of the bot
+     *
+     * @return The bot role, or null if no role matches
+     */
+    @Nullable
+    default Role getRoleByBot(@Nonnull String userId)
+    {
+        return getRoleByBot(MiscUtil.parseSnowflake(userId));
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/Guild.java:1617]: Missing throws

**Pull request merged**